rocev2/test/src/scala/rdma/RecvQTest.scala

```diff
@@ -0,0 +1,193 @@
+package rdma
+
+import org.scalatest.funsuite.AnyFunSuite
+import rdma.RdmaConstants.PSN_WIDTH
+import spinal.core._
+import spinal.core.sim._
+import spinal.lib._
+
+import scala.collection.mutable
+import scala.util.Random
+
+class SeqOutTest extends AnyFunSuite {
+
+  class RxDriver(
+      rx: Stream[Fragment[RdmaDataBus]],
+      clockDomain: ClockDomain,
+      opCode: OpCode.OpCode,
+      fragments: Int
+  ) {
+    val psnQueue = new mutable.Queue[Int]
+    rx.valid #= false
+    rx.last #= false
+    rx.bth.opcodeFull #= opCode.id
+
+    def push(psnList: List[Int]): Unit = {
+      psnQueue ++= psnList
+    }
+
+    def clear(): Unit = {
+      psnQueue.clear()
+    }
+
+    val driver = fork {
+      while (true) {
+        if (psnQueue.nonEmpty) {
+          rx.bth.psn #= psnQueue.dequeue()
+          rx.valid #= true
+          for (i <- 1 to fragments) {
+            if (i == fragments) {
+              rx.last #= true
+            }
+            clockDomain.waitSamplingWhere(rx.ready.toBoolean)
+            rx.last #= false
+          }
+        } else {
+          rx.valid #= false
+          clockDomain.waitRisingEdge()
+        }
+      }
+    }
+  }
+
+  class PsnSetter(
```

Comment:
this class has no member fields, consider refactor it to object

Reply:
It's a driver bind with io bundles. I think the class semantic is more suitable this case, I found that other io binders are define as class in SpinalHDL source.